DrivewayDentDeletion/Operators/configure-postgres.sh

```diff
@@ -12,29 +12,60 @@ function usage {
   echo "Usage: $0 -n <namespace>"
 }
 
-namespace="cp4i"
+NAMESPACE="cp4i"
 
 while getopts "n:r:" opt; do
   case ${opt} in
-    n ) namespace="$OPTARG"
+    n ) NAMESPACE="$OPTARG"
       ;;
     \? ) usage; exit
       ;;
   esac
 done
 
-namespace=$(echo $namespace | sed 's/-/_/g')
-postgresPod=$(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}')
+DB_POD=$(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}')
+DB_SVC="$(oc get cm postgres-config -ojson | jq '.data["postgres.env"] | split("\n  ")' | grep DATABASE_SERVICE_NAME | cut -d "=" -f 2- | tr -dc '[a-z0-9-]\n').postgres.svc.cluster.local"
+DB_USER=$(echo $NAMESPACE | sed 's/-/_/g')
+DB_NAME=db_${DB_USER}
+DB_PASS=$(LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 32 ; echo)
```

Comment:
LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 32 ; echo ?
Whats happening here

Reply:
random string generation for the password

Reply:
okay